app/views/contexts/summary_context.py

```diff
@@ -0,0 +1,48 @@
+from typing import Generator, Mapping, Optional, Union
+
+from app.questionnaire.location import Location
+
+from .context import Context
+from .section_summary_context import SectionSummaryContext
+
+
+class SummaryContext(Context):
+    def __call__(
+        self, answers_are_editable: bool = False, return_to: Optional[str] = None
+    ) -> dict[str, Union[str, list, bool]]:
+
+        groups = list(self._build_all_groups(return_to))
```

Comment:
This is an existing issue and not something that was introduced, only exposed as I wanted to reuse it. As agreed above With Andrew the whole  SectionSummaryContext needs a coat of looking at

Reply:
Umm alright, since this card if just about have the page, happy to refactor separately. Can you raise a card if one hasn't been already, so that we don't forget?

Reply:
Done, I have copied and pasted yours and Andrews comments into a card in tech